client/src/Stats.js

```diff
@@ -56,41 +54,12 @@ export default class Stats extends Component {
     super(props)
 
     this.state = {
-      loading: true,
       week: this.props.week,
-      stats: null
+      stats: this.props.stats
     }
   }
 
-  componentWillMount () {
-    this._fetchWeek(this.state.week)
-  }
-
-  componentWillReceiveProps (nextProps: Props) {
-    let week = nextProps.week
-
-    this.setState({
-      loading: true,
-      week: week,
-      stats: null
-    })
-
-    this._fetchWeek(week)
-  }
-
-  _fetchWeek (num: number) {
-    let url = `/weeks/${num}`
-    if (num === 0) url = '/stats'
-
-    $.get(url, (result) => {
-      let stats = result ? result.stats : {}
-      this.setState({ stats: stats, loading: false })
-    })
-  }
-
   render () {
-    if (this.state.loading) return (<Loading />)
```

Comment:
I love how easy this is to do with react. Boom my state comes from up the chain now. not my problem any more